libs/providers/flagd-web/src/lib/local-storage-flagd-cache.ts

```diff
@@ -0,0 +1,98 @@
+import { EvaluationContext, FlagValue, ResolutionDetails } from '@openfeature/js-sdk';
+import { Md5 } from 'ts-md5';
+import { FlagdCache } from './flagd-cache';
+
+type CacheItem<T = FlagValue> = {
+  expiry: number;
+  details: ResolutionDetails<T>;
+  variant: string;
+};
+
+const PREXFIX = '@OpenFeature/flagd-';
+const SEPARATOR = '--';
+const DEFAULT_OPTIONS = {
+  expiryCheckInterval: 60000,
+  itemTtl: 300,
+};
+
+export interface Options {
+  expiryCheckInterval: number;
+  itemTtl: number;
+}
+
+function getOptions(options: Partial<Options> = {}) {
+  return {
+    ...DEFAULT_OPTIONS,
+    ...options,
+  };
+}
+
+export class LocalStorageFlagdCache implements FlagdCache {
+  private _ttl: number | undefined;
+
+  constructor(options?: Partial<Options>) {
+    const defaultedOptions = getOptions(options);
+    this._ttl = defaultedOptions.itemTtl === 0 ? undefined : defaultedOptions.itemTtl;
+    setInterval(() => {
+      const currentTimestamp = new Date().getTime();
+
+      Object.entries(localStorage).forEach((each) => {
```

Comment:
This sort of iteration isn't ideal, but in this context I think it's reasonable.